vault/vault.go

```diff
@@ -38,7 +54,16 @@ func NewDefaultVault() EnvVault {
 		log.Warnf("Unable to load Environment vars: %s", err)
 	}
 	log.Infof("Vault address '%s' ", conf.Address)
+
 	vaultClient, _ := api.NewClient(conf)
+
+	if os.Getenv("VAULT_TOKEN") == "" {
```

Comment:
Just starting to look at that changes, and I have noticed, the api function [NewClient()](https://github.com/Nitro/sidecar-executor/blob/master/vendor/github.com/hashicorp/vault/api/client.go#L345) already lookup for the `VAULT_TOKEN` environment variable per [client.go#L378](https://github.com/Nitro/sidecar-executor/blob/master/vendor/github.com/hashicorp/vault/api/client.go#L378) and set the `token` parameter into `client.token` on the `client` instance. Here `vaultClient` above, I guess we can remove this `os.GetEnv` call.

Reply:
Well, fair point, but we actually need to know if the env var was set or not, as the source of truth. I _could_ call `client.Token()` here instead, but then it's quite opaque where the token came from. This way you can read the code and right away see what the logic hangs on.